go/api/options/script_options.go

```diff
@@ -0,0 +1,141 @@
+// Copyright Valkey GLIDE Project Contributors - SPDX Identifier: Apache-2.0
+
+package options
+
+import (
+	"sync"
+	"unsafe"
+)
+
+// #include "../../lib.h"
+import "C"
+
+// ScriptOptions represents options for script execution
+type ScriptOptions struct {
+	keys []string
+	args []string
+}
+
+// NewScriptOptions creates a new ScriptOptions with default values
+func NewScriptOptions() *ScriptOptions {
+	return &ScriptOptions{
+		keys: []string{},
+		args: []string{},
+	}
+}
+
+// WithKeys sets the keys for the script
+func (o *ScriptOptions) WithKeys(keys []string) *ScriptOptions {
+	o.keys = keys
+	return o
+}
+
+// WithArgs sets the arguments for the script
+func (o *ScriptOptions) WithArgs(args []string) *ScriptOptions {
+	o.args = args
+	return o
+}
+
+// GetKeys returns the keys for the script
+func (o *ScriptOptions) GetKeys() []string {
+	return o.keys
+}
+
+// GetArgs returns the arguments for the script
+func (o *ScriptOptions) GetArgs() []string {
+	return o.args
+}
+
+type ClusterScriptOptions struct {
+	*ScriptOptions
+	*RouteOption
+}
+
+// NewClusterScriptOptions creates a new ClusterScriptOptions with default values
+func NewClusterScriptOptions() *ClusterScriptOptions {
+	return &ClusterScriptOptions{
+		ScriptOptions: NewScriptOptions(),
+		RouteOption:   &RouteOption{},
+	}
+}
+
+// WithRouteOptions sets the route options for the cluster script
+func (o *ClusterScriptOptions) WithRouteOptions(routeOption *RouteOption) *ClusterScriptOptions {
+	o.RouteOption = routeOption
+	return o
+}
+
+// WithScriptOptions sets the script options for the cluster script
+func (o *ClusterScriptOptions) WithScriptOptions(scriptOptions *ScriptOptions) *ClusterScriptOptions {
```

Comment:
You need to ensure that keys and route are mutually exclusive. A user should have no option to set both.